File: go/500.600/504.base7.go
package main

import (
	"fmt"
)

func convertToBase7(num int) string {
	if num == 0 {
		return "0"
	}
	neg := num < 0
	if neg {
		num = -num
	}
	ret := []byte{}
	for num > 0 {
		ret = append(ret, byte(num%7)+'0')
		num /= 7
	}
	if neg {
		ret = append(ret, '-')
	}
	start, end := 0, len(ret)-1
	for start < end {
		ret[start], ret[end] = ret[end], ret[start]
		start++
		end--
	}
	return string(ret)
}

func main() {
	fmt.Println(convertToBase7(100), "202")
	fmt.Println(convertToBase7(-7), "-10")
}

File: 500.600/504.base7.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;

class base7
{
public:
    string convertToBase7(int num)
    {
        if (num == 0)
        {
            return "0";
        }
        bool neg = num < 0;
        num = abs(num);
        string ret = "";
        while (num > 0)
        {
            ret.push_back(num % 7 + '0');
            num /= 7;
        }
        if (neg)
        {
            ret.push_back('-');
        }
        reverse(ret.begin(), ret.end());
        return ret;
    }
};

int main(int argc, char const *argv[])
{
    base7 cls;
    cout << cls.convertToBase7(100) << " -> 202" << endl;
    cout << cls.convertToBase7(-7) << " -> -10" << endl;
    return 0;
}
